docs/src/proposals/program-recovery.md

```diff
@@ -0,0 +1,96 @@
+# Program Recovery
+
+## Problem
+
+As of Solana v1.11, the Upgradable Program Loader makes closing programs an irreversible action.
```

Comment:
Is this not the case in v1.10? I believe I know the answer to this, but this language is a little confusing, as it suggests there is a change in v1.11 that causes the issue.

Reply:
I was trying to express the time frame of this issue in a way that it makes sense even after the proposal is implemented.

How about "In Solana versions up to v1.??"? But I'm not sure if that even makes sense, since features get activated at epoch boundaries in the midst of a Solana release cycle.

We can also just remove that part

```suggestion
The Upgradable Program Loader makes closing programs an irreversible action.
```

Reply:
I think removing it is fine, since the merge date/commit of the proposal will act as a timestamp.
(I toyed with some different wordings to be more explicit, but haven't come up with anything non-clunky yet.)